tools/toolchains/arm.py

```diff
@@ -380,76 +381,67 @@ def __init__(self, target, *args, **kwargs):
         if not set(("ARM", "ARMC6")).intersection(set(target.supported_toolchains)):
             raise NotSupportedException("ARM/ARMC6 compiler support is required for ARMC6 build")
 
-        if target.core.lower().endswith("fd"):
-            self.flags['common'].append("-mcpu=%s" % target.core.lower()[:-2])
-            self.flags['ld'].append("--cpu=%s" % target.core.lower()[:-2])
-            self.SHEBANG += " -mcpu=%s" % target.core.lower()[:-2]
-        elif target.core.lower().endswith("fd-ns"):
-            self.flags['common'].append("-mcpu=%s" % target.core.lower()[:-5])
-            self.flags['ld'].append("--cpu=%s" % target.core.lower()[:-5])
-            self.SHEBANG += " -mcpu=%s" % target.core.lower()[:-5]
-        elif target.core.lower().endswith("f"):
-            self.flags['common'].append("-mcpu=%s" % target.core.lower()[:-1])
-            self.flags['ld'].append("--cpu=%s" % target.core.lower()[:-1])
-            self.SHEBANG += " -mcpu=%s" % target.core.lower()[:-1]
-        elif target.core.startswith("Cortex-M33"):
-            self.flags['common'].append("-mcpu=cortex-m33+nodsp")
-            self.flags['common'].append("-mfpu=none")
-            self.flags['ld'].append("--cpu=Cortex-M33.no_dsp.no_fp")
-        elif not target.core.startswith("Cortex-M23"):
-            self.flags['common'].append("-mcpu=%s" % target.core.lower())
-            self.flags['ld'].append("--cpu=%s" % target.core.lower())
-            self.SHEBANG += " -mcpu=%s" % target.core.lower()
-
-        if target.core == "Cortex-M4F":
+        core = target.core
+        if CORE_ARCH[target.core] == 8:
+            if (not target.core.endswith("-NS")) and kwargs.get('build_dir', False):
+                # Create Secure library
+                build_dir = kwargs['build_dir']
+                secure_file = join(build_dir, "cmse_lib.o")
+                self.flags["ld"] += ["--import_cmse_lib_out=%s" % secure_file]
+
+            # Add linking time preprocessor macro DOMAIN_NS
+            if target.core.endswith("-NS"):
+                define_string = self.make_ld_define("DOMAIN_NS", "0x1")
+                self.flags["ld"].append(define_string)
+                core = target.core[:-3]
+            else:
+                # Add secure build flag
+                self.flags['cxx'].append("-mcmse")
+                self.flags['c'].append("-mcmse")
+
+        cpu = {
+            "Cortex-M0+": "cortex-m0plus",
+            "Cortex-M4F": "cortex-m4",
+            "Cortex-M7F": "cortex-m7",
+            "Cortex-M7FD": "cortex-m7",
+            "Cortex-M33": "cortex-m33+no_dsp+no_fp",
+            "Cortex-M33F": "cortex-m33+no_dsp",
+            "Cortex-M33FD": "cortex-m33"}.get(core, core)
+
+        cpu = cpu.lower()
+        self.flags['common'].append("-mcpu=%s" % cpu)
+        self.SHEBANG += " -mcpu=%s" % cpu
+
+        # FPU handling
+        if core == "Cortex-M4F":
```

Comment:
This, and all `elif` clauses look very much like a lookup table. Could we lookup-table-ify these too?

Reply:
I am not comfortable converting floating point settings to tables, as not all settings are covered in all if/else statement and we have some outliers.
Will leave this to experts for later

Reply:
Agreed on the benefits of a lookup table here. But that can be done in a follow up pr.